src/main/java/com/derongan/minecraft/deeperworld/extensions/DeeperWorldExtensions.kt

```diff
@@ -31,4 +32,21 @@ internal fun Player.getLeashedEntities(): List<LivingEntity> {
     return getNearbyEntities(20.0, 20.0, 20.0)
         .filterIsInstance<LivingEntity>()
         .filter { it.isLeashed && it.leashHolder == this }
+}
+
+internal fun Player.teleportWithSpectator(loc : Location){
+    val nearbyPlayers = getNearbyEntities(5.0, 5.0, 5.0).filterIsInstance<Player>()
+
+    var spectator : Player? = null
```

Comment:
Since multiple people can be spectators of the same player, I think we should just filter the list of nearby players and run a forEach on them.

When selecting just one, I'd prefer `nearbyPlayers.first { it.spectatorTarget == this }`